include/bl_storage_internal.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BL_STORAGE_INTERNAL_H_
+#define BL_STORAGE_INTERNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @defgroup bl_storage Bootloader storage (protected data).
+ * @{
+ */
+
+const struct monotonic_counter *get_counter_struct(uint16_t description);
+
+/** This library implements monotonic counters where each time the counter
+ *  is increased, a new slot is written.
+ *  This way, the counter can be updated without erase. This is, among other things,
+ *  necessary so the counter can be used in the OTP section of the UICR
+ *  (available on e.g. nRF91 and nRF53).
+ */
+struct monotonic_counter {
```

Comment:
Isn't that changes (for CONFIG_NRFX_NVMC=y) the layout in OTP?

Reply:
Intention was to only change slot size and not description and slot number

Reply:
So I tired to change this back to how it was originally but it seems my OTP halfword read function is either non functional or something is wrong at least it can't read the monotonic counter without these old changes. I don't think I'll have time to clean it up.